api/gateway_services.proto

```diff
@@ -76,6 +76,23 @@ service GatewayRegistry {
       delete: "/gateways/{gateway_id}"
     };
   };
+
+  // SetGatewayLBSSecret is used to encrypt a plaintext value and store it along with the gateway.
+  // Requires the RIGHT_GATEWAY_WRITE_SECRET right.
+  rpc StoreGatewaySecret(StoreGatewaySecretRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      patch: "/gateways/{gateway_ids.gateway_id}/secret"
+      body: "*"
+    };
+  };
```

Comment:
So there is still one secret per gateway entity, which gets patched in its entirety here?

How would we support multiple secrets in the future?